### PR TITLE
Add Source NAT list on VPC

### PR DESCRIPTION
**Pull request by ddegoede**
This PR adds a source NAT list on a VPC. Next to the VPC CIDR also all CIDRs in the source NAT list will be added to the source NAT in iptables. This allows traffic over the private gateway from other VPCs to be source NATted using the source NAT IP address to this VPC.

When adding a new VPC the source NAT list can entered here:
image

The source NAT list is in the form of a comma separated list of CIDRs e.g: 
- 10.0.0.0/8 
- 192.168.22.0/24,192.168.45.39/32

The list can also be edited after creation:
image

Adding a source NAT list will result in a vr.json to be sent to the router, from that vr.json a data bag called virtualrouter.json will be created:
image

And IP tables will contain entries in the SOURCE_NAT_LIST chain:
image

### CODE REVIEW

File: cosmic-core/db-scripts/src/main/resources/db/schema-534to535.sql [hunk 1]
```diff
@@ -5,3 +5,6 @@
 -- Manufacturer field
 ALTER table guest_os ADD `manufacturer_string` varchar(64) DEFAULT "Mission Critical Cloud" COMMENT 'String to put in the Manufacturer field in the XML of a KVM VM';
 
+-- Add source nat list
+ALTER TABLE `cloud`.`vpc` ADD COLUMN `source_nat_list` varchar(255) DEFAULT NULL COMMENT 'List of CIDRS to source NAT on VPC' AFTER `redundant`;
```
Comment (borisroman) [inline cosmic-core/db-scripts/src/main/resources/db/schema-534to535.sql]: @ddegoede We just release 5.3.5.0. So this will need to go to 5.3.6.0.

**Review by borisroman — DISMISSED** [review]

**Comment by ddegoede** [comment]
When adding a new VPC the source NAT list can entered here:
image

The source NAT list is in the form of a comma separated list of CIDRs e.g: 
- 10.0.0.0/8 
- 192.168.22.0/24,10.192.168.45.39/32

The list can also be edited after creation:
image

Adding a source NAT list will result in a vr.json to be sent to the router, from that vr.json a data bag called virtualrouter.json will be created:
image

And IP tables will contain entries in the SOURCE_NAT_LIST chain:
image

File: cosmic-core/db-scripts/src/main/resources/db/schema-534to535.sql (hunk 1 above)
Comment (remibergsma) [reply cosmic-core/db-scripts/src/main/resources/db/schema-534to535.sql]: Upgrade path is just merged

**Comment by ddegoede** [comment]
Since @eliasgomes and others are rewriting all testing, no test have been added to this PR. As discussed this is the test scenario we should add for this PR:

- create a default VPC (1st) with no tiers;
- add a source NAT list with the VPC CIDR of the second VPC;
- add a private network;
- add a private gateway to the first VPC with a default allow ACL;
- add a static route on the first VPC for the second VPC CIDR pointing to the second VPC;
- create an internal VPC (2nd);
- add a private gateway to the second VPC with a default allow ACL;
- add a default static route to second VPC pointing at the first VPC;
- add a tier with a default allow ACL;
- add a VM to the tier;
- test a ping to 192.168.23.1 from the VM;

If the ping succeeds, test is successful.

**Pull request merged**